pgmpy/models/BayesianModel.py

```diff
@@ -1030,3 +1030,16 @@ def get_markov_blanket(self, node):
         blanket_nodes = set(blanket_nodes)
         blanket_nodes.remove(node)
         return list(blanket_nodes)
+    
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            network_struct1 = (self.nodes(),self.edges())
+            network_struct2 = (other.nodes(),other.edges())
+            cpd1 = self.cpds
+            cpd2 = other.cpds
+            if network_struct1 == network_struct2 and cpd1 == cpd2:
```

Comment:
As far as I remember `self.cpds` would be a list. So `cpd1 == cpd2` would fail if the same CPDs are added to the network in a different order.

Reply:
@ankurankan I am stuck at resolving this part, do you have some suggestion on how to resolve this part? I tried to do something along the lines of converting the cpd objects to string (I understand that this doesn't make much sense) and then sorting it, the other alternative I came up with is checking every `cpd` of `self` to see if it is in the `other.cpds` list but this would be very in-efficient. I will appreciate some help in this part.